packages/block-editor/src/components/rich-text/index.js

```diff
@@ -503,8 +503,13 @@ PrivateRichText.isEmpty = ( value ) => {
  * @see https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/rich-text/README.md
  */
 const PublicForwardedRichTextContainer = forwardRef( ( props, ref ) => {
-	const context = useBlockEditContext();
-	const isPreviewMode = context[ isPreviewModeKey ];
+	const { isPreviewMode } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+
+		return {
+			isPreviewMode: getSettings().isPreviewMode,
+		};
+	}, [] );
```

Comment:
This adds an extra store subscription, I think using the context is a performance optimization.